analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -286,3 +286,15 @@ private fun Collection<Edge>.toPackageReferenceForest(
 
     return nodes.values.filter { it.incomingEdges.isEmpty() }.map { getPackageReference(it.id) }.toSortedSet()
 }
+
+// See https://golang.org/ref/mod#pseudo-versions.
+private val PSEUDO_VERSION_REGEX = "^v0.0.0-([\\d]{14}-([0-9a-f]*)$)".toRegex()
```

Comment:
How about using both a non-capturing and a named group, like

    "^v0.0.0-(?:[\\d]{14}-(?<sha1>[0-9a-f]+)$)"

Note that I also replaced `*` with `+`.